core/network/impl/router_libp2p.cpp

```diff
@@ -113,60 +155,98 @@ namespace kagome::network {
         });
   }
 
+  void RouterLibp2p::handleTransactionsProtocol(
+      std::shared_ptr<Stream> stream) const {
+    auto rw = std::make_shared<libp2p::basic::MessageReadWriterUvarint>(stream);
+    rw->read([wself{weak_from_this()}, stream, rw](auto read_result) {
+      auto self = wself.lock();
+      if (!self) return;
```

Comment:
Is this a normal situation, shouldn't it be logged?

Reply:
It's normal, in the case of async operations on the exit for example.